core/src/main/java/org/apache/iceberg/MetricsConfig.java

```diff
@@ -195,6 +198,16 @@ private static MetricsMode sortedColumnDefaultMode(MetricsMode defaultMode) {
     }
   }
 
+  private static MetricsMode parseMode(String modeString, MetricsMode fallback, String context) {
+    try {
+      return MetricsModes.fromString(modeString);
+    } catch (IllegalArgumentException err) {
+      // User override was invalid, log the error and use the default
+      LOG.warn("Ignoring invalid metrics mode ({}): {}", context, modeString, err);
```

Comment:
This may not log err?  (It looks like the parameterized version of log.warn)

Reply:
If the last argument is a Throwable, the logger does the right thing.